src/App.js

```diff
@@ -1,13 +1,30 @@
 import BookList from './views/Books/BookList';
+import { BrowserRouter, Switch, Route } from 'react-router-dom';
+import BookDetail from './views/Books/BookDetail';
+import Main from './views/Main/Main';
 import './App.css';
 
 function App() {
   // TODO: Add routes to books & views
   return (
-    <main className="container">
-      <h1>Library Catalog</h1>
-      <BookList />
-    </main>
+    <BrowserRouter>
+      <main className="container">
+        <h1>Library Catalog</h1>
+        <Switch>
+          <Route exact path = '/books'>
+            <BookList />
+          </Route>
+
+          <Route exact path = '/books/:id'>
+            <BookDetail />
+          </Route>
+
+          <Route exact path = '/'>
+            <Main/>
+          </Route>
+        </Switch>
+      </main>
```

Comment:
When we call the main component, remember to add a route around the component which points to  the location (to='/')